webapp/src/helpers/__tests__/useSocket.ts

```diff
@@ -12,7 +12,7 @@ describe('useSocket', () => {
     const socket1 = useSocket();
     expect(mockIo).toHaveBeenCalledWith(process.env.REACT_APP_API_ORIGIN);
     const socket2 = useSocket();
-    expect(socket1).toEqual(socket2);
+    expect(socket1).toBe(socket2);
```

Comment:
Why is `toBe` used instead of something like `StrictEqual`?

Reply:
The contract of the function is that is returns the exact same object on subsequent calls, not just an object with the same type and structure. The only thing that matters for this test case is object identity. Two objects with the same type and structure but are different instances should fail this test, but "equal"/"strict equal" would still pass.

Reply:
the toEqual method deep compare the arrays 
tooBe method. This checks for reference
strictEqual() method tests if two values are equal, using the === operator

Reply:
Thanks for clarifying.  So, is memoization here is used in order to use the same socket instance, and not just purely for efficiency like in the classic Fibonacci sequence problem?

Reply:
In this situation, it's actually to prevent side effects. Every time `io` is called, a new connection is established, but the app only needs one in total. So, this is lazy loading a resource, not memoization. Memoization usually entails storing computed values keyed by the arguments, but this function takes no arguments.